src/main/java/org/elasticsearch/app/api/server/IndexerController.java

```diff
@@ -57,46 +58,61 @@ public String getKibanaHost() {
 
     @PutMapping(path = "/configs", consumes = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.ACCEPTED)
+    @ExceptionHandler
     public String saveConfig(@RequestBody String jsonConfig) {
         River river = configManager.save(jsonConfig);
         return river.getRiverName();
     }
 
     @PutMapping(path = "/configAndIndex", consumes = MediaType.APPLICATION_JSON_VALUE)
     @ResponseStatus(HttpStatus.ACCEPTED)
+    @ExceptionHandler
     public String saveConfigAndStart(@RequestBody String jsonConfig) {
-        String id = saveConfig(jsonConfig);
-        startIndex(id);
-        return id;
+        String indexName = saveConfig(jsonConfig);
+        startIndex(indexName);
+        return indexName;
     }
 
-    @PostMapping("/configs/{id}/start")
+    @PutMapping(path = "/{oldIndexName}/_rename/{newIndexName}")
```

Comment:
Yes, it is case 1. This is just bad by design. It originates in having indexName as ID (changed in this PR to artificial ID). So the suggested path is `/configs/{configID}/rename` and newname in requestbody? or just path `/configs/{oldIndexName}/rename/{newIndexName}`?

Reply:
You should not use `rename`. "[2. Use Nouns Instead of Verbs in Endpoints](https://www.freecodecamp.org/news/rest-api-best-practices-rest-endpoint-design-examples/)". 


 I would use one of the following alternatives:


On get for `/configs/{configID}` i would return json:
```
{
   id: 123
   name: "old name"
   ....
}
```

With PUT `/configs/{configID}` you can use new json:
```
{
   id: 123
   name: "new name"
   ...
}
```
---------------

Alternative to that is use of:
GET `/configs/{configID}/name`  returning  "old name"
PUT `/configs/{configID}/name` with input string "new name"

Reply:
Implemented PATCH `/configs/{configID}/name` with input string "new name"